runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/middleware/AwsSigningMiddleware.kt

```diff
@@ -147,17 +149,28 @@ public class AwsSigningMiddleware(private val config: Config) : ModifyRequestMid
                 contextHashSpecification != null -> contextHashSpecification
                 config.isUnsignedPayload -> HashSpecification.UnsignedPayload
                 body is HttpBody.Empty -> HashSpecification.EmptyBody
-                body is HttpBody.Streaming && !body.isReplayable -> {
-                    coroutineContext.warn<AwsSigningMiddleware> {
-                        "unable to compute hash for unbounded stream; defaulting to unsigned payload"
+                body is HttpBody.Streaming && (!body.isReplayable || body.isEligibleForAwsChunkedStreaming()) -> {
```

Comment:
**Nit**: Seems like the `isReplayable` check could be moved into `isEligibleForAwsChunkedStreaming`.